config/routes.rb

```diff
@@ -1,23 +1,34 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root to: 'users#index'
+  devise_for :users, controllers: { confirmations: 'confirmations' } 
 
   resources :users, only: [:index, :show, :create] do
-    resources :posts, only: [:new, :show, :index, :create] do
-      resources :comments, only: [:new, :create]
+    resources :posts, only: [:new, :show, :index, :create, :destroy] do
+      resources :comments, only: [:new, :create, :destroy]
       member do
         put "like", to: "posts#like"
         delete "unlike", to: "likes#destroy"
       end
     end
   end
 
-  resources :posts, only: [] do
+  resources :posts, only: [:new, :show, :index, :create, :destroy] do
     member do
       get "like", to: "posts#like", as: :like_post
       get "unlike", to: "posts#unlike", as: :unlike_post
     end
   end
+  delete "posts/:id/remove", to: "posts#destroy", as: "remove_post"
+  delete "users/:id/posts/:id/comments/:id/remove", to: "comments#destroy", as: "remove_comment"
+
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: [:index] do
+        resources :posts, only: [:index, :show] do
+          resources :comments, only: [:index, :create]
+        end
+        resources :comments, only: [:index]
+      end
+    end
+  end
```

Comment:
- [ Optional] Please it would be better to apply the dry principle, you can just use the code from lines 24 to 33 to set your routes there is no need from repeating the same routes also make sure to add all the methods you used in posts, likes and comments and users after this.